### ISSUE TITLE
TClass::GetStreamerInfo crashes for several classes

### ISSUE THREAD

**Issue by linev** [root-project/root#8194]
To reproduce error, just call:
```
TClass::GetClass("TTreeCache")->GetStreamerInfo()
```
Produces crash with output:
```
input_line_9:6:31: error: 'MissCache' is a protected member of 'TTreeCache'
   default_delete<TTreeCache::MissCache> _1;
                              ^
/home/linev/build/json/include/TTreeCache.h:85:11: note: declared protected here
   struct MissCache {
          ^
input_line_9:7:16: error: 'MissCache' is a protected member of 'TTreeCache'
   TTreeCache::MissCache* _0;
               ^
/home/linev/build/json/include/TTreeCache.h:85:11: note: declared protected here
   struct MissCache {
          ^
Error in <Load>: Could not declare TEmulatedTuple<TTreeCache::MissCache*,default_delete<TTreeCache::MissCache> >

 *** Break *** segmentation violation

===========================================================
There was a crash.
This is the entire stack trace of all threads:
===========================================================
```

There are also few other classes with similar problems:
```
TClass::GetClass("TMVA::VarTransformHandler")->GetStreamerInfo();
TClass::GetClass("TTreeCacheUnzip")->GetStreamerInfo();
```

Probably, issue https://github.com/root-project/root/issues/7276 connected with this one.

**Comment by guitargeek** [comment]
> Probably, issue #7276 connected with this one.

Doesn't seem like it, because issue #7276 got fixed in 6.24 but this issue can still be reproduced in ROOT master.

**Comment by ferdymercury** [comment]
PR here: https://github.com/root-project/root/pull/15876
